Added ability to save out overlay textures using the TextureSaveDialog.

rkhoury opened the pull request:
This is part of the work specified by #586, allowing the
ability to save out the overlay in the TextureViewer.  If no overlays
are on then there is no option to save the overlay. Currently there is
no option to remap the overlay to a grayscale or absolute value range
before saving. This can be a future task.

NOTE: the overlay texture resource that's saved out is not the blended
texture that the user will see in the TextureViewer, it is just the
overlay itself. The ability to save out the blended texture would be a
future task.

Commit:
Added ability to save out overlay textures using the TextureSaveDialog.

This is part of the work specified by github Issue 586, allowing the
ability to save out the overlay in the TextureViewer.  If no overlays
are on then there is no option to save the overlay. Currently there is
no option to remap the overlay to a grayscale or absolute value range
before saving. This can be a future task.

NOTE: the overlay texture resource that's saved out is not the blended
texture that the user will see in the TextureViewer, it is just the
overlay itself. The ability to save out the blended texture would be a
future task.

Commit:
Missing Qt UI file change from previous commits.

baldurk commented on qrenderdoc/Windows/TextureViewer.cpp:
```diff
@@ -3377,11 +3377,19 @@ void TextureViewer::on_saveTex_clicked()
   }
 
-  TextureSaveDialog saveDialog(*texptr, config, this);
+  const auto overlayTexID = m_Output->GetDebugOverlayTexID();
```
No `auto` please unless the type is really horrible like an iterator or lambda, `ResourceId` is preferred here :smile:.

rkhoury replied:
No worries - I'm happy to make this change.  I tend to use auto when I can easily see the definition implied by assignment (it's use is within 5 lines of code or so).

baldurk commented on qrenderdoc/Windows/TextureViewer.cpp:
```diff
@@ -3377,11 +3377,19 @@ void TextureViewer::on_saveTex_clicked()
 
-  TextureSaveDialog saveDialog(*texptr, config, this);
+  const auto overlayTexID = m_Output->GetDebugOverlayTexID();
+  const bool hasSelectedOverlay = (m_Output->GetTextureDisplay().overlay != DebugOverlay::NoOverlay);
```
I'd prefer not to add a query function like `GetTextureDisplay` when the information should be available already. Can you just use `m_TexDisplay` here?

rkhoury replied:
Oh man!! I totally didn't see that member. The joys of learning a new codebase...

baldurk reviewed (changes requested):
This looks good, I've made a couple of minor comments on the code.

I've fixed the CI issues on v1.x now, so when you fix those if you rebase onto HEAD again it should be a known good version.

rkhoury committed:
Removing unnecessary addition of the ReplayOutput::GetTextureDisplay.

This member function wasn't required and the TextureDisplay was
accessible from within the TexxtureViewer as m_TexDisplay.

Commit:
Missed a file change from the previous commit.
